File: iOS/EvianBlog/api/writings/ArticleAPI.swift
//
//  article-api.swift
//  EvianBlog
//
//  Created by Evian张 on 2020/5/18.
//  Copyright © 2020 Evian张. All rights reserved.
//

import Alamofire

import Foundation
import Combine

extension BlogAPI {
	func getArticlesCount() -> AnyPublisher<UInt, BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Article.getArticlesCount)
	}

	func getArticleMetas(pageIndex: UInt, pageSize: UInt) -> AnyPublisher<[ArticleMeta], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Article.getArticleMetas(pageIndex, pageSize))
	}

	func getArticle(title: String) -> AnyPublisher<Article, BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Article.getArticle(title))
	}

	func getArticleTitles() -> AnyPublisher<[String], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Article.getArticleTitles)
	}
}

File: Android/app/src/main/java/top/evian_zhang/evianblog/api/writings/ArticleAPI.kt
package top.evian_zhang.evianblog.api.writings

import top.evian_zhang.evianblog.Article
import top.evian_zhang.evianblog.ArticleMeta
import top.evian_zhang.evianblog.api.BlogAPI
import top.evian_zhang.evianblog.api.Endpoint

suspend fun BlogAPI.getArticlesCount(): Int {
    return this.fetch(Endpoint.Writings.Article.getArticleCount)
}

suspend fun BlogAPI.getArticleMetas(pageIndex: Int, pageSize: Int): Array<ArticleMeta> {
    return this.fetch(Endpoint.Writings.Article.getArticleMetas(pageIndex, pageSize))
}

suspend fun BlogAPI.getArticle(title: String): Article {
    return this.fetch(Endpoint.Writings.Article.getArticle(title))
}

suspend fun BlogAPI.getArticleTitles(): Array<String> {
    return this.fetch(Endpoint.Writings.Article.getArticleTitles)
}
